src/gsi/screen_to_ncdiag.f90

```diff
@@ -0,0 +1,30 @@
+module screen_to_ncdiag
```

Comment:
@RussTreadon-NOAA I started looking into this option, but there is at least one challenge.  How should missing values be defined in ncdiag?  That library doesn't have an `r_missing` variable.  We could define one for each of the data types supported, but how would we remain consistent between the GSI, EnKF, and ncdiag?

This also brings up a side note.  I picked `r_missing` not knowing if that was the best option.  I noticed several places where apparent missing values of -99999.0, -999.0, and others were used.  Should `<kind>_missing=-9999_<kind>` be the standard?

Reply:
Good points.  Let's hear what others have to say.

Reply:
@RussTreadon-NOAA After reading through this again and thinking about it, I believe the best solution is to move this subroutine to ncdiag.  I've opened issue NOAA-EMC/GSI-ncdiag#1 and will have a PR opened shortly.  As for the missing value, netCDF4 defines a missing value as `NF90_FILL_FLOAT`, which is already used when writing 2d data in nc_diag, so I will go ahead with that as the replacement for `NaN`.

Reply:
@RussTreadon-NOAA This is worrisome.  These should not be getting printed out.  Looking at lines [1269-1275](https://github.com/NOAA-EMC/GSI-ncdiag/blob/395d6dd869697d049e8f41ab7ab68fad257ea4fc/ncdiag/ncdw_metadata.F90#L1269-L1275) of ncdw_metadata.f90, `*` should be a valid operator, so that error should not be printed:

```
               case("*")
                  if(isnan(metadata_operand_x) .or. isnan(metadata_operand_y)) then
                     f_metadata_value = NLAYER_FILL_FLOAT
                  else
                     f_metadata_value = &
                        sngl(metadata_operand_x * metadata_operand_y)
                  endif
```

I will see if I can replicate these warnings and debug.

Reply:
I see the issue.  Using a string length of 2 for the input operator to the `nc_diag_metadata_to_single_operator` subroutine results in the second character being filled with the next byte of memory if that operator is only length 1.  A character length of `*` should be used instead.  I had caught this locally, but never committed it, which is why it worked in testing previously. 

I'll make the change to ncdiag, test it, and open a PR.